ptbr/tests/test_training_cli.py

```
@@ -0,0 +1,960 @@
+        valid_cfg["model"]["totally_unknown_param"] = True
+        result = validate_config(valid_cfg)
+        assert result.ok  # extra keys are warnings, not errors
+        assert any("totally_unknown_param" in w for w in result.warnings)
+
+    def test_none_for_optional_is_ok(self, valid_cfg: dict) -> None:
+        valid_cfg["model"]["labels_encoder"] = None
+        result = validate_config(valid_cfg)
+        assert result.ok
+
+    def test_sensitive_values_are_redacted(self, valid_cfg: dict) -> None:
+        valid_cfg.setdefault("environment", {})["hf_token"] = "hf_secret_token"
```

Ruff (0.15.1)

[error] Possible hardcoded password assigned to: "hf_token"

(S105)